### PR TITLE
Use origin as custom app configuration audience on decode

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Fix staging - use origin from CORS as custom app configuration token audience value on decode.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/routers/search.py [hunk 1]
```diff
@@ -140,19 +140,23 @@ def search_documents(
     the search database. The continuation token can be used to get the next set of
     results from the search database. See the request schema for more details.
     """
+    origin = request.headers.get("origin")
+    if origin is not None:
```
Comment (diversemix) [inline app/api/api_v1/routers/search.py:144]: Assume if origin is None then we'll throw on 159?

Comment (diversemix) [reply app/api/api_v1/routers/search.py:144]: We should perhaps test this as part of `test_decoding_token_with_invalid_aud_raises_expired_signature_token_error` ?

**Review by diversemix — APPROVED** [review]
Looks good - but let's pop in that test ?

File: app/api/api_v1/routers/search.py (hunk 1 above)
Comment (katybaulch) [reply app/api/api_v1/routers/search.py:144]: If the audience is None here, decode_config_token will throw an `InvalidAudienceError` which is captured under `test_search_decoding_token_raises_PyJWTError`

Comment (diversemix) [reply app/api/api_v1/routers/search.py:144]: That test confirms the http response given a thrown error. Not ... if the origin is none the error is thrown

Or am I missing something?

Comment (katybaulch) [reply app/api/api_v1/routers/search.py:144]: That's fair, I'll patch that in a new test :)

**Pull request merged**